packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -254,6 +255,7 @@ export async function processHTTPRequest<TContext>(
     // in ApolloServer#graphQLServerOptions, before runHttpQuery is invoked).
     const context = cloneObject(options.context);
     return {
+      parent: httpRequest,
```

Comment:
The same `httpRequest` that you are attaching here is already present on the `request` object on a property called `http`, in a pattern that is almost a mirror image of where it's found on the `response` object which is seen a few lines down from here.

In other words, it's already present on the `request` that's passed into this `buildRequestContext` function since it's populated in `parseGraphQLRequest` here and added to `request` prior to `buildRequestContext` being called:

https://github.com/apollographql/apollo-server/blob/2728b9be5e1f84ea0abebcc97bb60082f82fd8ed/packages/apollo-server-core/src/runHttpQuery.ts#L416

### `parseGraphQLRequest`:

https://github.com/apollographql/apollo-server/blob/2728b9be5e1f84ea0abebcc97bb60082f82fd8ed/packages/apollo-server-core/src/runHttpQuery.ts#L284